### PR TITLE
use expo av instead of old audio recorder. dep using expo-permissions

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] audio perms ios
- [x] audio perms android
- [x] location perms ios
- [x] location perms android
- [x] contacts perms ios
- [x] contacts perms android
- [x] qr scan perms ios
- [x] qr scan perms android
- [x] go file cleanup ios 
- [x] go file cleanup android

### CODE REVIEW

File: shared/actions/platform-specific/index.native.tsx [hunk 1]
```diff
@@ -62,61 +62,42 @@ const requestPermissionsToWrite = async () => {
 
 export const requestAudioPermission = async () => {
   let chargeForward = true
-  // TODO use expo-av etc and unify around that
-  const Permissions = require('expo-permissions') as typeof PermissionsType
```
Comment (chrisnojima) [inline shared/actions/platform-specific/index.native.tsx:66]: using expo-permissions is deprecated. instead we use the module specific permissions checks

File: shared/actions/json/settings.json [hunk 2]
```diff
@@ -66,7 +66,7 @@
       "enabled": "boolean"
     },
     "loadedContactPermissions": {
-      "status": ["'granted'", "'never_ask_again'", "'undetermined'"]
+      "status": ["'granted'", "'denied'", "'undetermined'"]
```
Comment (chrisnojima) [inline shared/actions/json/settings.json:69]: use actual value from the library

File: shared/actions/platform-specific/index.native.tsx [hunk 3]
```diff
@@ -578,10 +509,8 @@ const manageContactsCache = async (
     if (state.settings.contacts.waitingToShowJoinedModal && resolved) {
       actions.push(SettingsGen.createShowContactsJoinedModal({resolved}))
     }
-  } catch (error) {
-    if (!(error instanceof RPCError)) {
```
Comment (chrisnojima) [inline shared/actions/platform-specific/index.native.tsx:582]: these are typically not RPCerrors

File: shared/app/index.native.tsx [hunk 4]
```diff
@@ -54,7 +54,19 @@ const NativeEventsToRedux = () => {
   return null
 }
 
+// dont' remake engine/store on reload
```
Comment (chrisnojima) [inline shared/app/index.native.tsx:57]: stop hot loading from making a new engine / reregistering the listeners

File: shared/util/native-modules.native.tsx [hunk 5]
```diff
@@ -1,5 +1,5 @@
-import {NativeModules as RNNativeModules} from 'react-native'
-import {isIOS} from '../constants/platform.native'
+import {NativeModules as RNNativeModules, Platform} from 'react-native'
+export const isIOS = Platform.OS === 'ios'
```
Comment (chrisnojima) [inline shared/util/native-modules.native.tsx:2]: stop circular import

File: go/chat/attachments/uploader.go [hunk 6]
```diff
@@ -247,7 +247,11 @@ func (u *Uploader) clearOldUploaderTempDirs(ctx context.Context, delay time.Dura
 		u.Debug(ctx, "clearOldUploaderTempDirs: cleaning: %s", dir)
 		os.RemoveAll(dir)
 	}
-	if !u.G().IsMobileAppType() {
+	if u.G().IsMobileAppType() {
+		dir := filepath.Join(u.G().GetCacheDir(), "../AV")
```
Comment (chrisnojima) [inline go/chat/attachments/uploader.go:251]: cleanup expo av audio file leftovers if they exist

File: shared/chat/audio/audio-recorder.native.tsx [hunk 7]
```diff
@@ -56,17 +56,14 @@ const useVisible = (reduxVisible: boolean, dragX: SVN, dragY: SVN) => {
 const useRecording = (conversationIDKey: Types.ConversationIDKey) => {
   const ampTracker = React.useRef(new AmpTracker()).current
   const ampScale = useSharedValue(0)
-  const unifyAmp = (amp: number) => {
-    return isIOS ? 10 ** (amp * 0.05) : Math.min(1.0, amp / 22000)
```
Comment (chrisnojima) [inline shared/chat/audio/audio-recorder.native.tsx:60]: this is the same value for android and ios now

File: shared/chat/conversation/info-panel/settings/index.tsx [hunk 8]
```diff
@@ -50,7 +50,6 @@ const SettingsPanel = (props: SettingsPanelProps) => {
   )
 
   const onShowClearConversationDialog = () => {
-    dispatch(Chat2Gen.createNavigateToThread({conversationIDKey, reason: 'misc'}))
```
Comment (chrisnojima) [inline shared/chat/conversation/info-panel/settings/index.tsx:53]: we don't need this and its racy on android

File: go/chat/attachments/uploader.go [hunk 9]
```diff
@@ -232,6 +232,20 @@ func (u *Uploader) Complete(ctx context.Context, outboxID chat1.OutboxID) {
 	u.clearTempDirFromOutboxID(ctx, outboxID)
 }
 
+func (u *Uploader) clearExpoAudioPaths(ctx context.Context) {
```
[the remark this answers is not in the document]
Comment (zoom-ua) [reply go/chat/attachments/uploader.go:235]: lgtm

**Pull request merged**